### DISCUSSION TITLE
Markor needs a tips and tricks FAQ or Wiki

### DISCUSSION THREAD

**Discussion by nutpantz** [gsantner/markor#2397] [General]
i just found this posted in an issue... its a great tip or trick when using markor  
but i would never have found it if i did not go down a rabbit hole cause i was bored.

> Here is a fun hack for you
> 
> Markor will execute any JS code in a script tags when you switch to view mode
> 
> I use the following code to open a file with today's date in the current folder. If the file does not exist it will be created.
> 
> You can modify this to open a new file in the current folder with any programmatically generated name. If you want a new unique name you could use seconds or a UUID
> 
> ```
> <script>
> let anchor = document.createElement('a');
> 
> // This asinine approach is from https://stackoverflow.com/a/28149561
> // offset in milliseconds
> let offset = (new Date()).getTimezoneOffset() * 60000;
> 
> let iso = (new Date(Date.now() - offset)).toISOString()
>     
> let today = iso.split('T')[0]
> 
> let textNode = document.createTextNode("Today's worklog");
> 
> // Append the textNode as a child to anchor.
> anchor.appendChild(textNode);
> 
> anchor.href = "./" + today + ".md";
> 
> anchor.style.fontSize = "x-large";
> 
> document.body.appendChild(anchor);
> 
> window.location.href = anchor.href
> 
> </script>
> ```
now i have to go down another rabbit hole trying to modify it to add hours + minutes to the filename
which is likely explained somewhere else in the issues section..
 
a tips and tricks section would be great 
( as many of us have NO IDEA how scripts/templates work and examples would be great if they were in one place at the top of the page) 

then i need to know if i can do this auto name with a template.... more rabbit hole hunting

**Comment by gsantner** [comment]
Average users are not programmers. Markor not is a web development teaching app. If users want to use such things, they need to learn it somewhere else, Markor can't be doing everything. This is advanced stuff for users that know about it.

Also by feeling average users don't create custom tokenized templates but maybe rather simple files. The tokens I myself don't even use @harshad1 wrote some nice comment here somewhere describing it (which woulf fit to the doc folder/docs).

**Comment by nutpantz** [comment]
true average users have no advanced skills, but im willing to bet if there was a list of scripts what worked for markor there would be a lot of average users who would cut-paste and use them..  
ive never seen that script before and even i was able to google  how to change it to my needs.

**Comment by harshad1** [comment]
Can we pin a discussion? If so I can link several such things there

- Todo txt query language
- snippet syntax
- hidden / unknown features (long press checkbox action in markdown)
- some neat JavaScript based applets within markor

**Reply by gsantner** [reply]
Possibly but only 1 or 2, generally said ""the good stuff"" is git versioned - in the /doc/ folder. For the rest of "FAQ" I consider generally  discussions like this one here, with the benefit that that it's also good searchable.

**Reply by nutpantz** [reply]
> Can we pin a discussion? If so I can link several such things there
> 
>     * Todo txt query language
> 
>     * snippet syntax
> 
>     * hidden / unknown features (long press checkbox action in markdown)
> 
>     * some neat JavaScript based applets within markor

post what you can even here.. just start a new topic for scripts and say no help is offered and no warranty is implied then hopefully no one will blame you when their microwave dies.
